shogun-lib/src/main/java/de/terrestris/shogun/lib/service/CacheService.java

```diff
@@ -43,4 +47,23 @@ public void evictCache() throws Exception {
 
         cache.evictAllRegions();
     }
+
+    public void evictCacheRegions(String... region) throws Exception {
+        if (region == null) {
+            return;
+        }
+        if (entityManager == null) {
+            throw new Exception("Could not get the entity manager.");
+        } else {
+            EntityManagerFactory entityManagerFactory = entityManager.getEntityManagerFactory();
+            SessionFactory sessionFactory = entityManagerFactory.unwrap(SessionFactory.class);
+            Cache cache = sessionFactory.getCache();
+            for (String r : region) {
+                if (StringUtils.isEmpty(r)) {
+                    continue;
+                }
+                cache.evictRegion(r);
```

Comment:
What happens for an unavailable region identifier? Do we need to check if the region exists before?

Reply:
good point. I'll do some testing

Reply:
A NullPointerException is thrown if one region identifier is invalid, preventing the following regions from being cleared. AFAIK there is no hibernate method which checks if an identifier is valid or not.

I could wrap `cache.evictRegion(r);` into a try-catch block. That way, other (possible) valid regions will still be cleared.

Reply:
implemented here: 9cfa96669971e5da3e06bdba9a5a1270dd3bc7b2